### PR TITLE
Sync promotions UI

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/0/1207344337317797/f 

### Description
UI for the sync promotions in bookmarks and passwords screens.

### Bookmarks
**Promo doesn't show if no bookmarks saved**
- [x] Fresh install. Visit Bookmarks; verify no promo is showing

**Promo shows only if not dismissed before**
- [x] Visit a site and add as a bookmark. Visit Bookmarks; verify sync promo shows
- [x] Dismiss the promo; verify it disappears
- [x] Leave the screen and re-enter; verify it doesn't come back

**Promo sync path**
- [x] Fresh install, add a bookmark and open Bookmarks screen
- [x] Tap **Set up Sync**
- [x] Return **without** setting up sync; verify promo still showing
- [x] Tap **Set up Sync** again
- [x] This time, set up sync
- [x] Return to Bookmarks screen and verify the promo is **not** visible

### Passwords
**Promo doesn't show if no passwords saved**
- [x] Fresh install. Visit Passwords screen; verify no promo is showing

**Promo shows only if not dismissed before**
- [x] Add a password (manually or by accepting autofill's offer to save a login)
- [x] Return to password screen; verify promo appears
- [x] Dismiss the promo; verify it disappears
- [x] Leave the screen and re-enter; verify it doesn't come back

**Promo sync path**
- [x] Fresh install, add a password and open Passwords screen
- [x] Tap **Set up Sync**
- [x] Return **without** setting up sync; verify promo still showing
- [x] Tap **Set up Sync** again
- [x] This time, set up sync
- [x] Return to Passwords screen and verify the promo is **not** visible

### CODE REVIEW

**Comment by CDRussell** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4892?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4892** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4892?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4891** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4891?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/di/AutofillModule.kt [hunk 1]
```diff
@@ -154,3 +155,6 @@ class AutofillModule {
  */
 @ContributesPluginPoint(scope = AppScope::class, boundType = AutofillFragmentResultsPlugin::class)
 interface UnusedAutofillResultPlugin
+
+@ContributesPluginPoint(scope = AppScope::class, boundType = PasswordsScreenPromotionPlugin::class)
```
Comment (karlenDimla) [inline autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/di/AutofillModule.kt:159]: Can’t this be `ActivityScope`?

File: saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/di/SavedSitesModule.kt [hunk 2]
```diff
@@ -139,3 +141,6 @@ class SavedSitesModule {
         return database.syncMetadataDao()
     }
 }
+
+@ContributesPluginPoint(scope = AppScope::class, boundType = BookmarksScreenPromotionPlugin::class)
```
Comment (karlenDimla) [inline saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/di/SavedSitesModule.kt:145]: Can this also be `ActivityScope`?

**Review by karlenDimla — APPROVED** [review]

**Comment by CDRussell** [comment]
### Merge activity

* **Aug 29, 11:58 AM EDT**: @CDRussell started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/4892).

**Pull request merged**